models/vehicle.rb

```diff
@@ -203,31 +203,48 @@ def total_work_time_in_range(range_start, range_end)
 
       if @total_work_time_in_range[[range_start, range_end]].nil? # if info for this range is not already calculated, calculate
         total_work_time = 0
-        working_day_indices_in_range(range_start, range_end).group_by{ |range_day| range_day.modulo(7) }.each{ |group|
-          week_day_index = group[0]
-          occurence = group[1].size
-          # TODO : fix case where serveral timewindows correspond to the same day
-          tw_index = working_week_days.find_index(week_day_index)
-          tw = self.timewindow
-          tw ||= self.sequence_timewindows[tw_index]
-          total_work_time += (tw.end - tw.start) * occurence
-        }
-        @total_work_time_in_range[[range_start, range_end]] = total_work_time
+        if self.timewindow.nil? && self.sequence_timewindows.empty?
+          total_work_time = working_day_indices_in_range(range_start, range_end).size * [2**32, self.duration].compact.min
```

Comment:
This would work right?
```suggestion
          total_work_time = working_day_indices_in_range(range_start, range_end).size * (self.duration || 2**32)
```